tests/bitRangeReadPlugin.xlmap

```diff
@@ -0,0 +1,70 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes" ?>
+<logicalNameMap>
+    <variable name="SimpleScalar">
+        <type>int16</type>
+        <value>0</value>
+    </variable>
+    <variable name="F32">
+        <type>float32</type>
+        <value>0</value>
+    </variable>
+
+    <redirectedRegister name="Middle">
+        <targetDevice>this</targetDevice>
+        <targetRegister>SimpleScalar</targetRegister>
+        <plugin name="bitRange">
+            <parameter name="shift">3</parameter>
+            <parameter name="numberOfBits">10</parameter>
+        </plugin>
+    </redirectedRegister>
+
+    <redirectedRegister name="LoByte">
+        <targetDevice>this</targetDevice>
+        <targetRegister>SimpleScalar</targetRegister>
+        <plugin name="bitRange">
+            <parameter name="shift">0</parameter>
+            <parameter name="numberOfBits">8</parameter>
+        </plugin>
+    </redirectedRegister>
+    <redirectedRegister name="HiByte">
+        <targetDevice>this</targetDevice>
+        <targetRegister>SimpleScalar</targetRegister>
+        <plugin name="bitRange">
+            <parameter name="shift">8</parameter>
+            <parameter name="numberOfBits">8</parameter>
+        </plugin>
+    </redirectedRegister>
+    <redirectedRegister name="MidByte">
```

Comment:
Your docu says overlapping bit ranges are unspecified behaviour. Yet many of your tests are based on it.

Reply:
The documentation is missing an "in TransferGroups"

Reply:
Hm, that is not good. The configuration in the logical name mapping is "far away" from the application code, hence the decision to have overlapping bit ranges is done in a very different place then the decision to use a TransferGroup. This is error prone. We need at least a logic_error exception if someone tries to put overlapping bit ranges into a TransferGroup. Better would be working support for that...

Reply:
> Better would be working support for that...

How could working support for overlapping write logically work at all? The bit from which accessor would win in the group? The same is true for normal accessors to the same register. If you have the same register twice in a transfer group and use it for writing, behaviour is undefined.

Reply:
Ok, the documentation was massively incomplete, this is about writing, and only because it isn't clear which one wins. Making the group read-only on overlapping segments sounds like the way to go

Reply:
Adapted as discussed, overlapping ranges will turn the transfergroup read-only